### PR TITLE
update changelog ahead of v0.35 release

### PR DESCRIPTION
**Pull request by williambanfield**
This change moves the changelog entries from CHANGELOG_PENDING.md to CHANGELOG.md ahead of the 0.35 release.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,168 @@
...
+  - [ABCI] \#6408 Change the `key` and `value` fields from `[]byte` to `string` in the `EventAttribute` type. (@alexanderbez)
+  - [ABCI] \#5447 Remove `SetOption` method from `ABCI.Client` interface
+  - [ABCI] \#5447 Reset `Oneof` indexes for  `Request` and `Response`.
+  - [ABCI] \#5818 Use protoio for msg length delimitation. Migrates from int64 to uint64 length delimiters.
+  - [ABCI] \#3546 Add `mempool_error` field to `ResponseCheckTx`. This field will contain an error string if Tendermint encountered an error while adding a transaction to the mempool. (@williambanfield)
+  - [Version] \#6494 `TMCoreSemVer` has been renamed to `TMVersion`.
+	- It is not required any longer to set ldflags to set version strings
+  - [abci/counter] \#6684 Delete counter example app
+
+- P2P Protocol
```
Comment (tychoish) [inline CHANGELOG.md:41]: should we say "NONE" here just to be clear?

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,6 +1,168 @@
...
+  - [mempool] \#6466 The original mempool reactor has been versioned as `v0` and moved to a sub-package under the root `mempool` package.
+	Some core types have been kept in the `mempool` package such as `TxCache` and it's implementations, the `Mempool` interface itself
+	and `TxInfo`. (@alexanderbez)
+  - [crypto/sr25519] \#6526 Do not re-execute the Ed25519-style key derivation step when doing signing and verification.  The derivation is now done once and only once.  This breaks `sr25519.GenPrivKeyFromSecret` output compatibility. (@Yawning)
+  - [types] \#6627 Move `NodeKey` to types to make the type public.
+  - [config] \#6627 Extend `config` to contain methods `LoadNodeKeyID` and `LoadorGenNodeKeyID`
+  - [blocksync] \#6755 Rename `FastSync` and `Blockchain` package to `BlockSync`
+	(@cmwaters)
+
+- Blockchain Protocol
```
Comment (tychoish) [inline CHANGELOG.md:81]: should explicitly say none here

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,6 +1,168 @@
...
+  - [cli/indexer] \#6676 Reindex events command line tooling. (@JayT106)
+
+### FEATURES
+
+- [config] Add `--mode` flag and config variable. See [ADR-52](https://github.com/tendermint/tendermint/blob/master/docs/architecture/adr-052-tendermint-mode.md) @dongsam
+- [rpc] \#6329 Don't cap page size in unsafe mode (@gotjoshua, @cmwaters)
+- [pex] \#6305 v2 pex reactor with backwards compatability. Introduces two new pex messages to
+  accomodate for the new p2p stack. Removes the notion of seeds and crawling. All peer
+  exchange reactors behave the same. (@cmwaters)
+- [crypto] \#6376 Enable sr25519 as a validator key
```
Comment (tychoish) [inline CHANGELOG.md:99]: format? type?

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,6 +1,168 @@
...
+- [mempool] \#6466 Introduction of a prioritized mempool. (@alexanderbez)
+  - `Priority` and `Sender` have been introduced into the `ResponseCheckTx` type, where the `priority` will determine the prioritization of
+  the transaction when a proposer reaps transactions for a block proposal. The `sender` field acts as an index.
+  - Operators may toggle between the legacy mempool reactor, `v0`, and the new prioritized reactor, `v1`, by setting the
+  `mempool.version` configuration, where `v1` is the default configuration.
+  - Applications that do not specify a priority, i.e. zero, will have transactions reaped by the order in which they are received by the node.
+  - Transactions are gossiped in FIFO order as they are in `v0`.
+- [config/indexer] \#6411 Introduce support for custom event indexing data sources, specifically PostgreSQL. (@JayT106)
+- [fastsync/event] \#6619 Emit fastsync status event when switching consensus/fastsync (@JayT106)
```
Comment (tychoish) [inline CHANGELOG.md:108]: should we rename fastsync here to blocksync?

File: CHANGELOG.md (hunk 1 above)
Comment (cmwaters) [reply CHANGELOG.md:41]: Or we can just remove the entry

File: CHANGELOG.md (hunk 4 above)
Comment (cmwaters) [reply CHANGELOG.md:108]: Yeah good idea

File: CHANGELOG.md (hunk 3 above)
Comment (williambanfield) [reply CHANGELOG.md:99]: Type seems most apt. "derived using the sr25519" is maybe a little verbose.

File: CHANGELOG.md (hunk 4 above)
Comment (williambanfield) [reply CHANGELOG.md:108]: Done in all places, good catch.

File: CHANGELOG.md (hunk 1 above)
Comment (williambanfield) [reply CHANGELOG.md:41]: Just going to delete the section.

**Review by tychoish — APPROVED** [review]

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,6 +1,163 @@
 # Changelog
 
-Friendly reminder, we have a [bug bounty program](https://hackerone.com/tendermint).
+Friendly reminder: We have a [bug bounty program](https://hackerone.com/tendermint).
+
+## v0.35
+Special thanks to external contributors on this release: @JayT106, @bipulprasad, @alessio, @Yawning, @silasdavis,
```
Comment (creachadair) [inline CHANGELOG.md:6]:
```suggestion

Special thanks to external contributors on this release: @JayT106, @bipulprasad, @alessio, @Yawning, @silasdavis,
```

**Review by creachadair — APPROVED** [review]
Only one entirely-unnecessary cosmetic nit. 🙂

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,6 +1,163 @@
...
+- [state/privval] \#6578 No GetPubKey retry beyond the proposal/voting window (@JayT106)
+- [rpc] \#6615 Add TotalGasUsed to block_results response (@crypto-facs)
+- [cmd/tendermint/commands] \#6623 replace `$HOME/.some/test/dir` with `t.TempDir` (@tanyabouman)
+- [statesync] \6807 Implement P2P state provider as an alternative to RPC (@cmwaters)
+
+### BUG FIXES
+
+- [privval] \#5638 Increase read/write timeout to 5s and calculate ping interval based on it (@JoeKash)
+- [blockchain/v1] [\#5701](https://github.com/tendermint/tendermint/pull/5701) Handle peers without blocks (@melekes)
+- [blockchain/v1] \#5711 Fix deadlock (@melekes)
```
Comment (marbar3778) [inline CHANGELOG.md:155]: v1 was deleted,
```suggestion
```

File: CHANGELOG.md [hunk 7]
```diff
@@ -1,6 +1,163 @@
...
+- [evidence] \#6375 Fix bug with inconsistent LightClientAttackEvidence hashing (cmwaters)
+- [rpc] \#6507 Ensure RPC client can handle URLs without ports (@JayT106)
+- [statesync] \#6463 Adds Reverse Sync feature to fetch historical light blocks after state sync in order to verify any evidence (@cmwaters)
+- [blocksync] \#6590 Update the metrics during blocksync (@JayT106)
+- [gitignore] \#6668 Fix gitignore of abci-cli (@tanyabouman)
```
Comment (marbar3778) [inline CHANGELOG.md:160]:
```suggestion
```
? not really needed

File: CHANGELOG.md [hunk 8]
```diff
@@ -1,6 +1,163 @@
...
+  - [rpc] \#6019 standardise RPC errors and return the correct status code (@bipulprasad & @cmwaters)
+  - [rpc] \#6168 Change default sorting to desc for `/tx_search` results (@melekes)
+  - [cli] \#6282 User must specify the node mode when using `tendermint init` (@cmwaters)
+  - [state/indexer] \#6382 reconstruct indexer, move txindex into the indexer package (@JayT106)
+  - [cli] \#6372 Introduce `BootstrapPeers` as part of the new p2p stack. Peers to be connected on  startup (@cmwaters)
+  - [config] \#6462 Move `PrivValidator` configuration out of `BaseConfig` into its own section. (@tychoish)
+  - [rpc] \#6610 Add MaxPeerBlockHeight into /status rpc call (@JayT106)
+  - [blocksync/rpc] \#6620 Add TotalSyncedTime & RemainingTime to SyncInfo in /status RPC  (@JayT106)
+  - [rpc/grpc] \#6725 Mark gRPC in the RPC layer as deprecated.
+  - [blockchain/v2] \#6730 Fast Sync v2 is deprecated, please use v0
```
Comment (marbar3778) [inline CHANGELOG.md:26]: nit:  these blockchain entries be renamed to blocksync

**Comment by marbar3778** [comment]
how is the upgrading.md file looking?

File: CHANGELOG.md [hunk 9]
```diff
@@ -1,6 +1,161 @@
...
+  - [config] \#6462 Move `PrivValidator` configuration out of `BaseConfig` into its own section. (@tychoish)
+  - [rpc] \#6610 Add MaxPeerBlockHeight into /status rpc call (@JayT106)
+  - [blocksync/rpc] \#6620 Add TotalSyncedTime & RemainingTime to SyncInfo in /status RPC  (@JayT106)
+  - [rpc/grpc] \#6725 Mark gRPC in the RPC layer as deprecated.
+  - [blocksync/v2] \#6730 Fast Sync v2 is deprecated, please use v0
+  - [rpc] Add genesis_chunked method to support paginated and parallel fetching of large genesis documents.
+  - [rpc/jsonrpc/server] \#6785 `Listen` function updated to take an `int` argument, `maxOpenConnections`, instead of an entire config object. (@williambanfield)
+  - [rpc] \#6820 Update RPC methods to reflect changes in the p2p layer, disabling support for `UnsafeDialPeers` and `UnsafeDialPeers` when used with the new p2p layer, and changing the response format of the peer list in `NetInfo` for all users.
+  - [cli] \#6854 Remove deprecated snake case commands. (@tychoish)
+- Apps
```
Comment (cmwaters) [inline CHANGELOG.md:32]:
```suggestion

- Apps
```

File: CHANGELOG.md [hunk 10]
```diff
@@ -1,6 +1,161 @@
...
+  - [ABCI] \#5818 Use protoio for msg length delimitation. Migrates from int64 to uint64 length delimiters.
+  - [ABCI] \#3546 Add `mempool_error` field to `ResponseCheckTx`. This field will contain an error string if Tendermint encountered an error while adding a transaction to the mempool. (@williambanfield)
+  - [Version] \#6494 `TMCoreSemVer` has been renamed to `TMVersion`.
+	- It is not required any longer to set ldflags to set version strings
+  - [abci/counter] \#6684 Delete counter example app
+
+- Go API
+  - [pubsub] \#6634 The `Query#Matches` method along with other pubsub methods, now accepts a `[]abci.Event` instead of `map[string][]string`. (@alexanderbez)
+  - [p2p] \#6618 Move `p2p.NodeInfo` into `types` to support use of the SDK. (@tychoish)
+  - [p2p] \#6583 Make `p2p.NodeID` and `p2p.NetAddress` exported types to support their use in the RPC layer. (@tychoish)
```
Comment (cmwaters) [inline CHANGELOG.md:45]: Can this be merged?

File: CHANGELOG.md [hunk 11]
```diff
@@ -1,6 +1,161 @@
...
+- [abci/client] \#5673 `Async` requests return an error if queue is full (@melekes)
+- [mempool] \#5673 Cancel `CheckTx` requests if RPC client disconnects or times out (@melekes)
+- [abci] \#5706 Added `AbciVersion` to `RequestInfo` allowing applications to check ABCI version when connecting to Tendermint. (@marbar3778)
+- [blocksync/v1] \#5728 Remove blocksync v1 (@melekes)
+- [blocksync/v0] \#5741 Relax termination conditions and increase sync timeout (@melekes)
+- [cli] \#5772 `gen_node_key` output now contains node ID (`id` field) (@melekes)
+- [blocksync/v2] \#5774 Send status request when new peer joins (@melekes)
+- [consensus] \#5792 Deprecates the `time_iota_ms` consensus parameter, to reduce the bug surface. The parameter is no longer used. (@valardragon)
+- [store] \#5888 store.SaveBlock saves using batches instead of transactions for now to improve ACID properties. This is a quick fix for underlying issues around tm-db and ACID guarantees. (@githubsands)
+- [consensus] \#5987 Remove `time_iota_ms` from consensus params. Merge `tmproto.ConsensusParams` and `abci.ConsensusParams`. (@marbar3778)
```
Comment (cmwaters) [inline CHANGELOG.md:128]: we should probably merge the two changes to `time_iota_ms`

File: CHANGELOG_PENDING.md [hunk 12]
```diff
@@ -8,158 +8,9 @@ Friendly reminder: We have a [bug bounty program](https://hackerone.com/tendermi
 
 ### BREAKING CHANGES
 
-- CLI/RPC/Config
```
Comment (cmwaters) [inline CHANGELOG_PENDING.md:11]: Do we want to keep these sub-categories

**Review by cmwaters — APPROVED** [review]
Few comments but mostly looks good to me.

File: CHANGELOG.md (hunk 11 above)
Comment (williambanfield) [reply CHANGELOG.md:128]: Done.

File: CHANGELOG.md (hunk 10 above)
Comment (williambanfield) [reply CHANGELOG.md:45]: Yes, Just checked. These were all were moved into the `types` package. Merged the entries.
#6583 and #6618

File: CHANGELOG_PENDING.md (hunk 12 above)
Comment (williambanfield) [reply CHANGELOG_PENDING.md:11]: yes! They were in v0.34 as well in the `CHANGELOG_PENDING.md`. Re-added

**Pull request merged**